platform/ios/src/MGLMapView.mm

```diff
@@ -3003,10 +3069,10 @@ - (void)locationManager:(__unused CLLocationManager *)manager didUpdateLocations
 
     if (! oldLocation || ! CLLocationCoordinate2DIsValid(oldLocation.coordinate) || [newLocation distanceFromLocation:oldLocation])
     {
-        self.userLocation.location = newLocation;
-
-        // deselect user if applicable since we don't do callout tracking yet
-        if ([self.selectedAnnotation isEqual:self.userLocation]) [self deselectAnnotation:self.userLocation animated:NO];
+        if (self.userTrackingState != MGLUserTrackingStateBegan)
+        {
+            self.userLocation.location = newLocation;
```

Comment:
The first location is also likely to be relatively inaccurate and get preempted quickly by another update. Don't think this is a blocker here, just another reason to do #3610.